src/test/java/mshantadze/SearchTest.java

```diff
@@ -1,35 +1,118 @@
 package mshantadze;
 
 import mshantadze.base.BaseTest;
+import mshantadze.pages.AdvancedSearchPage;
+import mshantadze.pages.CategoriesPage;
 import mshantadze.pages.HomePage;
 import mshantadze.pages.ProductListingPage;
+import mshantadze.utils.ui.enums.KeyWordOptions;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
+import java.math.BigDecimal;
 import java.util.List;
 
 public class SearchTest extends BaseTest {
     private static final Logger LOGGER = LoggerFactory.getLogger(SearchTest.class);
+    private final static String CATEGORIES_TITLE = "All Categories";
+    private final static String PRODUCTS_NOT_FOUND_MSG = "No exact matches found";
 
     @Test
     public void searchTest() {
+        WebDriver driver = createRemoteDriver();
+        driver.get(configInstance.get("url"));
         HomePage homePage = new HomePage(driver);
-        homePage.verifyThatPageIsLoaded();
+        homePage.verifyThatPageIsLoaded(driver);
         homePage.getSearchSection().typeInSearchField("Ball");
         homePage.getSearchSection().clickSearchButton();
         ProductListingPage plp = new ProductListingPage(driver);
-        List<WebElement> productNameLbls = plp.getProductNames();
-        for (WebElement productNameLbl : productNameLbls) {
-            String productName = productNameLbl.getText().toLowerCase();
-            LOGGER.info(productName);
-
-            Assert.assertTrue(
-                    productName.contains("ball"),
-                    "Expected element to contain word 'ball' but result was" + productName
-            );
+        Assert.assertTrue(plp.doesProductNameContain("Ball"));
+    }
+
+    @Test
+    public void emptySearch() {
+        WebDriver driver = createRemoteDriver();
+        driver.get(configInstance.get("url"));
+        HomePage homePage = new HomePage(driver);
+        homePage.verifyThatPageIsLoaded(driver);
+        homePage.getSearchSection().clickSearchButton();
+        CategoriesPage categoriesPage = new CategoriesPage(driver);
+        Assert.assertEquals(categoriesPage.getCategoriesTitle(), CATEGORIES_TITLE, "Element should have text \"All Categories\"");
+        Assert.assertTrue(driver.getCurrentUrl().contains("all-categories"), "url should contain all-categories");
+    }
+
+    @Test
+    public void noProductsFound() {
+        WebDriver driver = createRemoteDriver();
+        driver.get(configInstance.get("url"));
```

Comment:
Make method openPage() in your BasePage class. Make field String url for each page and set it in your constructor like 
public BasePage(WebDriver driver) {
 this.url = configInstance.get("url") + "/endpoint";
 }
 
 and after this somePage.openPage();